apps/src/StudioApp.js

```diff
@@ -1794,8 +1794,8 @@ StudioApp.prototype.configureDom = function (config) {
   // TODO (cpirich): make conditional for applab
   var belowViz = document.getElementById('belowVisualization');
   var referenceArea = document.getElementById('reference_area');
-  //TODO : this will get less ugly after we remove the flag - epeach
-  if ((config.app && (!config.app.includes('lab') || (!experiments.isEnabled('resourcesTab') && config.app.includes('lab'))) || !config.app) && referenceArea) {
+  // The check for instructionsWhenCollapsed checks for whether the app uses CSPTopInstructions
+  if ((!config.noInstructionsWhenCollapsed || !experiments.isEnabled('resourcesTab')) && referenceArea) {
```

Comment:
Even knowing what this does, I'm still struggling to convince myself that this is right - I think because I have no reason to associate `noInstructionsWhenCollapsed` with what's happening here in my brain. I wonder if doing something like this might make it slightly easier to parse:
```js
// noInstructionsWhenCollapsed is used by TopInstructions to determine whether we use CSPTopInstructions
// (in which case the instructions own the reference area) or CSFTopInstructions (in which case we want to
// append the referenceArea here).
const referenceAreaInTopInstructions = config.noInstructionsWhenCollapsed &&
  experiments.isEnabled('resourcesTab');
if (referenceArea && !referenceAreaInTopInstructions) {
 ```